linode/data_source_linode_stackscript.go

```diff
@@ -0,0 +1,149 @@
+				Description: "An array of Image IDs representing the Images that this StackScript is compatible for deploying with.",
+				Computed:    true,
+			},
+
+			"deployments_active": {
+				Type:        schema.TypeInt,
+				Description: "Count of currently active, deployed Linodes created from this StackScript.",
+				Computed:    true,
+			},
+			"user_gravatar_id": {
+				Type:        schema.TypeString,
+				Description: "The Gravatar ID for the User who created the StackScript.",
+				Computed:    true,
+			},
+			"deployments_total": {
+				Type:        schema.TypeInt,
+				Description: "The total number of times this StackScript has been deployed.",
+				Computed:    true,
+			},
+			"username": {
+				Type:        schema.TypeString,
+				Description: "The User who created the StackScript.",
+				Computed:    true,
+			},
+			"created": {
+				Type:        schema.TypeString,
+				Description: "The date this StackScript was created.",
+				Computed:    true,
+			},
+			"updated": {
+				Type:        schema.TypeString,
+				Description: "The date this StackScript was updated.",
+				Computed:    true,
+			},
+			"user_defined_fields": {
+				Description: "This is a list of fields defined with a special syntax inside this StackScript that allow for supplying customized parameters during deployment.",
+				Type:        schema.TypeList,
+				Computed:    true,
+				Optional:    true,
+				ConfigMode:  schema.SchemaConfigModeAttr,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"label": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"example": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"one_of": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"many_of": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"default": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceLinodeStackscriptRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(linodego.Client)
+	id := d.Get("id").(int)
+
+	ss, err := client.GetStackscript(context.Background(), id)
+	if err != nil {
+		return fmt.Errorf("Error getting Staakscript: %s", err)
+	}
+
+	if ss != nil {
+		d.SetId(strconv.Itoa(id))
+		d.Set("label", ss.Label)
+		d.Set("script", ss.Script)
+		d.Set("description", ss.Description)
+		d.Set("rev_note", ss.RevNote)
+		d.Set("is_public", ss.IsPublic)
+		d.Set("images", ss.Images)
+		d.Set("user_gravatar_id", ss.UserGravatarID)
+		d.Set("deployments_active", ss.DeploymentsActive)
+		d.Set("deployments_total", ss.DeploymentsTotal)
+		d.Set("username", ss.Username)
+		d.Set("created", ss.Created)
+		d.Set("updated", ss.Updated)
+		d.Set("user_defined_fields", ss.UserDefinedFields)
```

Comment:
I think this field is more complex than a simple `Set`.

Compare with https://github.com/terraform-providers/terraform-provider-linode/blob/02c1b76d2c3c1b93eee1ef822d896d0a6dffbc8c/linode/resource_linode_stackscript.go#L202-L221